language/vm/vm_runtime/src/runtime.rs

```diff
@@ -113,19 +114,60 @@ impl<'alloc> VMRuntime<'alloc> {
         txn_block: Vec<Transaction>,
         data_view: &dyn StateView,
     ) -> Vec<TransactionOutput> {
-        let txns = txn_block
-            .into_iter()
-            .map(|txn| {
-                SignedTransaction::try_from(txn)
-                    .expect("All transactions should be user transaction")
-            })
-            .collect();
-        execute_block(
-            txns,
-            &self.code_cache,
-            &self.script_cache,
-            data_view,
-            &self.publishing_option,
-        )
+        let mut result = vec![];
+        let blocks = chunk_block_transactions(txn_block);
+        for block in blocks {
+            match block {
+                TransactionBlock::UserTransaction(txns) => {
+                    result.append(&mut execute_user_transaction_block(
+                        txns,
+                        &self.code_cache,
+                        &self.script_cache,
+                        data_view,
+                        &self.publishing_option,
+                    ))
+                }
+                // TODO: Implement the logic for processing system transactions.
+                TransactionBlock::BlockPrologue(_) => unimplemented!(""),
+                TransactionBlock::WriteSet(_) => unimplemented!(""),
+            }
+        }
+        result
+    }
+}
+
+pub(crate) enum TransactionBlock {
```

Comment:
wonder if `TransactionBlock` specifically the use of Block is a good thing here given the confusion it seems to have caused and the correlation it suggests to a block in the executor sense... maybe `TransactionTypes`, `TransactionChunks`?